### ISSUE TITLE
Inadequate handling by Style/SafeNavigation cop

### ISSUE THREAD

**Issue by ashmaroli** [rubocop-hq/rubocop#6832]
## Expected behavior

Expected chained methods to be flagged by `Style/SafeNavigation` cop

## Actual behavior

Chained methods were not flagged. Only *non-chained* methods were flagged

## Steps to reproduce the problem

```ruby
# frozen_string_literal: true

# test.rb

def initialize(site, base, theme)
  @base_dir = if theme && theme.layouts_path.eql?(base)
                theme.root
              elsif site.theme && site.theme.layouts_path.eql?(base)
                site.theme.root
              else
                site.source
              end
end

def foo(base, theme)
  theme && theme.layouts_path.eql?(base)
end

def bar(base, theme)
  theme.layouts_path.eql?(base) if theme
end

theme && theme.layouts_path.eql?(base)
foo && foo.bar.nil?
foo && foo.bar
```
```
rubocop test.rb
Inspecting 1 file
C

Offenses:

test.rb:25:1: C: Style/SafeNavigation: Use safe navigation (&.) instead of checking if an object exists before calling the method.
foo && foo.bar
^^^^^^^^^^^^^^

1 file inspected, 1 offense detected
```

## RuboCop version

Include the output of `rubocop -V` or `bundle exec rubocop -V` if using Bundler. Here's an example:

```
$ [bundle exec] rubocop -V
0.65.0 (using Parser 2.6.0.0, running on ruby 2.4.5 x64-mingw32)
```

**Comment by rrosenblum** [comment]
Thanks for the bug report. 

The reason that only one piece of code was flagged for safe navigation is because all of the sample code contains calls to methods that `nil` implements. Safe navigation only works as a replacement when the method being called is not implemented by `nil`. For example `foo && foo.to_i` cannot be converted to `foo&.to_i`. Assuming `foo` is `nil`, in the first example `nil` is returned, in the second example `0` is returned. This also applies to method chains since the output can be changed.
